packages/@react-aria/i18n/src/useNumberParser.ts

```diff
@@ -10,39 +10,243 @@
+        return false;
+      }
+      return true;
+    }).map(p => p.value).join('');
+
+    // These are for replacing non-latn characters with the latn equivalent
+    let numerals = [...new Intl.NumberFormat(locale, {useGrouping: false}).format(9876543210)].reverse();
+    let indexes = new Map(numerals.map((d, i) => [d, i]));
+    let numeral = new RegExp(`[${numerals.join('')}]`, 'g');
+    let index = d => String(indexes.get(d));
+
+    return {minusSign, plusSign, decimal, group, currency, validCharacters, literals, numeral, index};
+  }, [formatter, intlOptions, locale]);
+
+  const parse = useCallback((value:string): number => {
+    // assuming a clean string
+    // to parse the number, we need to remove anything that isn't actually part of the number, for example we want '-10.40' not '-10.40 USD'
+    let fullySanitizedValue = value.replace(new RegExp(`[${symbols.literals}\\p{White_Space}]`, 'gu'), '');
+    // first match the prefix/suffix wrapper of everything that isn't a number
+    // then preserve that as we'll need to put it back together as the current input value
+    // with the actual numerals, parse them into a number
+    fullySanitizedValue = fullySanitizedValue.trim()
+      .replace(symbols.group, '')
+      .replace(symbols.decimal, '.')
+      .replace(symbols.numeral, symbols.index);
+
+    let newValue = fullySanitizedValue ? +fullySanitizedValue : NaN;
+    if (isNaN(newValue)) {
+      return NaN;
+    }
+
+    // accounting will always be stripped to a positive number, so if it's accounting and has a () around everything, then we need to make it negative again
+    if (intlOptions?.currencySign === 'accounting' && CURRENCY_SIGN_REGEX.test(value)) {
+      newValue = -1 * newValue;
+    }
+    // when reading the number, if it's a percent, then it should be interpreted as being divided by 100
+    if (intlOptions?.style === 'percent') {
+      newValue = newValue / 100;
+      // after dividing to get the percent value, javascript may get .0210999999 instead of .0211, so fix the number of fraction digits
+      newValue = +newValue.toFixed(intlOptions.maximumFractionDigits + 2);
+    }
+
+    return newValue;
+  }, [symbols, intlOptions]);
+
+  /**
+   * We could round by using toFixed and toPrecision with intlOptions fraction digits and significant digits
+   * respectively. However, it's hard to enforce the minimum's intlOptions also supports and it would take
+   * multiple trips from number to string and back again to accomplish.
+   * Instead, we format the number using the same format options that are passed in, strip it of everything
+   * that doesn't contribute to the numerical value, and parse it back to a number. This ensures the same
+   * rounding that the formatter will perform.
+   * @param value
+   */
+  let round = (value: number): number => {
+    let parts = formatter.formatToParts(value);
+    let strippedValue = parts.map((part) => {
```

Comment:
Why is this necessary? Doesn't the parser already handle stripping non-numeric parts? Do we need a separate function for this or would `numberParser.parse(formatter.formatToParts(value))` work?

Reply:
ah yep, this can work now, it was overly complicated from before, recent changes have made this possible

Reply:
though i assume you mean `parse(formatter.format(value));`